src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java

```diff
@@ -59,15 +67,25 @@
         connect(channels);
     }
 
-    protected void connect(Set<String> channels) {
+    protected void connect(Set<String> newChannels) {
         if (this.state != WSState.CONNECTED && this.state != WSState.CONNECTING) {
             log.debug("{} Connecting...", logPrefix);
             this.state = WSState.CONNECTING;
             this.webSocket = apiClient.createWebSocket(this.connectionRequest, this);
         } else {
             log.warn("{} Already connected", logPrefix);
         }
-        if (!channels.isEmpty()) subscribe(channels);
+        if (!newChannels.isEmpty()) {
+            this.subscribe(newChannels);
+            this.scheduledPingTask = this.getScheduler().scheduleAtFixedRate(new PingTask(newChannels),
+                    WEB_SOCKET_KEEP_ALIVE_TOPIC_PERIOD_SEC, WEB_SOCKET_KEEP_ALIVE_TOPIC_PERIOD_SEC, TimeUnit.SECONDS);
+        }
+    }
+    private ScheduledExecutorService getScheduler(){
+        if(isNull(this.scheduler)){
+            this.scheduler = Executors.newSingleThreadScheduledExecutor();
+        }
+        return this.scheduler;
     }
```

Comment:
why don't we have initialization in a constructor? It's not meant to be static field, so I don't see a reason for such initialization